### PR TITLE
WT-13415 Add a new eviction logging category eviction/WT_VERB_EVICTION

### PR DESCRIPTION
**Pull request by raviprakashgiri29**
This pull request adds a new eviction logging category "eviction".

### CODE REVIEW

**Comment by etienneptl** [comment]
Is the goal of this PR compared to the previous one to keep the other categories that we are deprecating but potentially used in the field?

**Comment by raviprakashgiri29** [comment]
> Is the goal of this PR compared to the previous one to keep the other categories that we are deprecating but potentially used in the field?

The following dependent changes will be implemented after these pull request changes are merged:

1. [SERVER-93723](https://jira.mongodb.org/browse/SERVER-93723): Modify the MongoDB server configuration to use the new "eviction" category instead of the older ones.
2. [WT-13416](https://jira.mongodb.org/browse/WT-13416): Remove the obsolete eviction logging categories ("evict", "evict_stuck" and "evictserver") from WiredTiger.

By implementing these changes in sequence, we will achieve a consolidated eviction logging mechanism across both WiredTiger and MongoDB.

**Comment by etienneptl** [comment]
Thank you for the explanation.
I think this set of changes should focus on adding the new category and not replacing for instance `WT_VERB_EVICTSERVER` to `WT_VERB_EVICTION` as this will break existing applications.
If you want to keep those changes in, then you need to update the verbose parsing to whenever `WT_VERB_EVICTSERVER` is enabled, then enable `WT_VERB_EVICTION` too.

File: src/conn/conn_api.c [hunk 1]
```diff
@@ -2333,11 +2333,12 @@ __wt_verbose_config(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig)
       {"checkpoint_progress", WT_VERB_CHECKPOINT_PROGRESS}, {"chunkcache", WT_VERB_CHUNKCACHE},
       {"compact", WT_VERB_COMPACT}, {"compact_progress", WT_VERB_COMPACT_PROGRESS},
       {"configuration", WT_VERB_CONFIGURATION}, {"error_returns", WT_VERB_ERROR_RETURNS},
-      {"evict", WT_VERB_EVICT}, {"evict_stuck", WT_VERB_EVICT_STUCK},
-      {"evictserver", WT_VERB_EVICTSERVER}, {"fileops", WT_VERB_FILEOPS},
-      {"generation", WT_VERB_GENERATION}, {"handleops", WT_VERB_HANDLEOPS}, {"log", WT_VERB_LOG},
-      {"history_store", WT_VERB_HS}, {"history_store_activity", WT_VERB_HS_ACTIVITY},
-      {"lsm", WT_VERB_LSM}, {"lsm_manager", WT_VERB_LSM_MANAGER}, {"metadata", WT_VERB_METADATA},
+      {"eviction", WT_VERB_EVICTION}, {"evict", WT_VERB_EVICT},
```
Comment (kommiharibabu) [inline src/conn/conn_api.c:2336]: Is `evict` must be the first and followed by `eviction`?

**Comment by raviprakashgiri29** [comment]
> Thank you for the explanation. I think this set of changes should focus on adding the new category and not replacing for instance `WT_VERB_EVICTSERVER` to `WT_VERB_EVICTION` as this will break existing applications. If you want to keep those changes in, then you need to update the verbose parsing to whenever `WT_VERB_EVICTSERVER` is enabled, then enable `WT_VERB_EVICTION` too.

I plan to merge these changes and [SERVER-93723](https://jira.mongodb.org/browse/SERVER-93723) at the same time. MongoDB uses only the "evict" category, and there are only 3 verbose messages in WiredTiger for WT_VERB_EVICT.

File: src/conn/conn_api.c (hunk 1 above)
Comment (raviprakashgiri29) [reply src/conn/conn_api.c]: Fixed in b7e97971c44ac62e068bd0722aba9c7f5c228a29

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by Sean04 — APPROVED** [review]
LGTM

**Pull request merged**